include/jemalloc/internal/edata.h

```diff
@@ -664,13 +664,7 @@ edata_cmp_summary_get(const edata_t *edata) {
 
 static inline int
 edata_cmp_summary_comp(edata_cmp_summary_t a, edata_cmp_summary_t b) {
-	int ret;
-	ret = (a.sn > b.sn) - (a.sn < b.sn);
-	if (ret != 0) {
-		return ret;
-	}
-	ret = (a.addr > b.addr) - (a.addr < b.addr);
-	return ret;
+  return (2 * ((a.sn > b.sn) - (a.sn < b.sn))) + ((a.addr > b.addr) - (a.addr < b.addr));
```

Comment:
nit: probably explain a bit what this is doing and why the (2*) is needed. maybe even keep the plain logic in the comment.

Reply:
Updated with descriptive comment explaining the logic.

Reply:
A few more nits:
1) the max line length should be 80 char
2) for multi-line comments, the format should be ([pasted from here](https://github.com/jemalloc/jemalloc/wiki/Coding-Style))
```
/*
 * This is a multi-line comment.  Follow sentence-ending periods with two
 * spaces.  Consider that we display code with fixed-width fonts; embrace the
 * old 20th-century convention and like it.
 */
```
3) Probably mention in the comments that, one factor that makes this optimization necessary is the fact that all branches here are common cases, i.e. it's hard to predict the outcome of the branches.
4) For consistency, maybe also apply the same trick to `edata_esnead_comp` below? In case it becomes more popular in the future.

Reply:
Thank you for the feedback. I have addressed your comments in the latest revision.